File: crates/klyron_formatter/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron_formatter {
FormatterBuilder::FormatterBuilder() : config_() {}
FormatterBuilder& FormatterBuilder::enabled(bool v) { config_.enabled = v; return *this; }
FormatterBuilder& FormatterBuilder::verbose(bool v) { config_.verbose = v; return *this; }
FormatterConfig FormatterBuilder::build() { return config_; }
}

File: crates/klyron_formatter/bindings/rust/builder.rs
use crate::config::FormatterConfig;

pub struct FormatterBuilder {
    config: FormatterConfig,
}

impl FormatterBuilder {
    pub fn new() -> Self { Self { config: FormatterConfig::default() } }
    pub fn build(self) -> FormatterConfig { self.config }
}

impl Default for FormatterBuilder {
    fn default() -> Self { Self::new() }
}
